src/construct.jl

```diff
@@ -436,7 +436,7 @@ function push_modex!(iter::ExprSplitter, mod::Module, ex::Expr)
         # Issue #427
         modifies_scope = false
         for a in ex.args
-            if isa(a, Expr) && a.head ∈ (:local, :global)
+            if Meta.isexpr(a, :local)
                 modifies_scope = true
                 break
             end
```

Comment:
Probably the right fix would be to add this `ex.head === :block` condition here.
```julia
        if ex.head === :block
            for a in ex.args
                if isa(a, Expr) && a.head ∈ (:local, :global)
                    modifies_scope = true
                    break
                end
            end
        end
```
Otherwise the same problem would still happen for `:toplevel` expression with `:local` declaration:
```julia
ex = Base.parse_input_line("""
local i
i = 42
println(i)
""")
```
, but we can safely ignore those expression in `:toplevel`.